src/app/content/highlights/utils/getHighlightLocationFiltersWithContent.ts

```diff
@@ -0,0 +1,15 @@
+import { HighlightLocationFilters, HighlightsTotalCountsPerPage } from '../types';
+import getHighlightLocationFilterForPage from './getHighlightLocationFilterForPage';
+
+export default (locationFilters: HighlightLocationFilters, totalCounts: HighlightsTotalCountsPerPage) => {
+
+  return Object.entries(totalCounts).reduce((result, [pageId]) => {
+    const location = getHighlightLocationFilterForPage(locationFilters, pageId);
+
+    if (location && !result.has(location.id)) {
+      result.set(location.id, location);
+    }
+
+    return result;
+  }, new Map() as HighlightLocationFilters);
```

Comment:
This could be `Set<LocationId>` since we are using it only for disabling Checkbox

Reply:
i did it this way to get the constant time access of Map.has, but it looks like Set.has does the same, so i'll change it